### ISSUE TITLE
AngleSharp does not correctly compute href when the base url has a port

### ISSUE THREAD

**Issue by meziantou** [AngleSharp/AngleSharp#501]
AngleSharp does not correctly compute href when the base url has a port:

```
var config = Configuration.Default.WithDefaultLoader();
var browsingContext = BrowsingContext.New(config);
var url = "http://localhost:5000";
var content = "<a id='sample' href='http://www.sample.com'></a>";

var htmlDocument = await browsingContext.OpenAsync(action => action.Address(url).Content(content));
var element = (IHtmlAnchorElement)htmlDocument.GetElementById("sample");
```
`element.Href` is `http://www.sample.com:5000/` instead of `http://www.sample.com`

**Comment by FlorianRappl** [comment]
Yes - you are right! Thanks for the report - will be fixed in the next version 👍

**Comment by FlorianRappl** [comment]
Fixed, see PR #503.
